### PR TITLE
Feat/loading

### PR DESCRIPTION
**Pull request by Tristan-C789**
This pull request introduces a loading state to various view models and screens, improving user experience by displaying a loading indicator during asynchronous operations. Additionally, it removes an unnecessary test case and refines the Firestore group creation logic.

## Loading state management:

* [`app/src/viewmodels/auth/SignInViewModel.tsx`](diffhunk://#diff-8bdfbf22908b8d8303c1c728d1dcaf66b93b1a3cb89b84675a42ccff72cf7495R23-R37): Added `isLoading` state and updated `handleSignIn` to manage loading state.
* [`app/src/viewmodels/auth/SignUpViewModel.tsx`](diffhunk://#diff-5bc45a9ad142784f4fcfa9499303570ab9b548accb4be38f2419db52566a14b9R32-R39): Added `isLoading` state and updated sign-up logic to manage loading state. 
* [`app/src/viewmodels/camera/CameraViewModel.tsx`](diffhunk://#diff-a205e961f414b954af3e44af3fb2d8cd23a685e766b2fb415ba2ae826eaa4695R141-R151): Added `isLoading` state and updated challenge creation logic to manage loading state.
* [`app/src/viewmodels/group/CreateGroupViewModel.tsx`](diffhunk://#diff-efc0cfe05d729aa4d8bc78ef0003b50bd3b881a3bcef4f4b1ebf47df4a86ea44R36): Added `isLoading` state and updated group creation logic to manage loading state. 

## UI updates to show loading indicators:

* [`app/src/views/auth/sign_in_screen.tsx`](diffhunk://#diff-22e265a24d4d8ecf63109fb0ea3066cbc7e84eefe48ccb7904fd1ebb23487e48R16): Added `LoadingSplash` component to display a loading indicator during sign-in. 
* [`app/src/views/auth/sign_up_screen.tsx`](diffhunk://#diff-a600fc46a262cb21e7b97fbd2c60c0b3d3e03f01b84bdb0da9f67f4c4da91ebdR11): Added `LoadingSplash` component to display a loading indicator during sign-up. 
* [`app/src/views/camera/CameraContainer.tsx`](diffhunk://#diff-1ce5a862054b4bcd7cbb954b41383975387dd068c955b68df3e1353c3e8462b6R9): Added `LoadingSplash` component to display a loading indicator during challenge creation. 
* [`app/src/views/group/CreateGroupScreen.tsx`](diffhunk://#diff-c720af7a690475f5a961dc11a02372c352422723165a74975bab7e4ad326115eR11): Added `LoadingSplash` component to display a loading indicator during group creation.

## Other changes:

* [`app/src/views/components/loading/loading_splash.tsx`](diffhunk://#diff-899af5266e004a096634b2af8098a3068ffbd1f7d7fced697877f6a2519b9f7eR1-R36): Created a new `LoadingSplash` component to display a loading indicator with customizable text.
* [`app/src/models/firebase/FirestoreCtrl.tsx`](diffhunk://#diff-fca90e0d4b4aa5847d4698253d5fe7d00902e8651597d25c08da4cad37d75a5eR726-L742): Simplified group creation logic by removing duplicate group check.
* [`__tests__/types/auth.test.tsx`](diffhunk://#diff-9e84b9d0a2b83a1ede5938d15dc36aaf1e55ce20660c2841527de6361331e797L320-L359): Removed an unnecessary test case for handling errors when creating a user in Firestore.

### CODE REVIEW

File: app/src/viewmodels/auth/SignUpViewModel.tsx [hunk 1]
```diff
@@ -54,6 +56,8 @@ export default function useSignUpViewModel(
       return;
     }
 
+    setIsLoading(true);
+    console.log("Signing up with email and password...");
```
Comment (EduardoNeville) [inline app/src/viewmodels/auth/SignUpViewModel.tsx:60]: remove logs if successful

File: app/src/viewmodels/auth/SignUpViewModel.tsx [hunk 2]
```diff
@@ -66,6 +70,8 @@ export default function useSignUpViewModel(
       console.error("Error during sign-up: ", error);
       alert("Failed to sign up. Please try again.");
     }
+    console.log("Sign up successful!");
```
Comment (EduardoNeville) [inline app/src/viewmodels/auth/SignUpViewModel.tsx:73]: idem previous comment

File: app/src/viewmodels/camera/CameraViewModel.tsx [hunk 3]
```diff
@@ -135,17 +136,20 @@ export default function useCameraViewModel(navigation: any, route: any) {
...
+        setIsLoading(false);
+
         if (!isInGroupArea(location, group)) {
           alert("You need to be in the group's area to create a challenge");
           navigation.navigate("GroupScreen", { currentGroup: group });
           return;
         }
       }
 
-      console.log("Picture URI: ", picture?.uri);
```
Comment (EduardoNeville) [inline app/src/viewmodels/camera/CameraViewModel.tsx:146]: just like that

File: app/src/viewmodels/home/MaximizeScreenViewModel.tsx [hunk 4]
```diff
@@ -127,13 +128,15 @@ export function useMaximizeScreenViewModel(
...
+      setIsLoading(true);
       await appendComment(commentData);
       setCommentList([...commentList, commentData]);
       setCommentText("");
+      setIsLoading(false);
     }
   };
 
-  const postDate: any = challenge.date ? challenge.date : new Date();
+  const postDate: Date = challenge.date ? challenge.date : new Date();
```
Comment (EduardoNeville) [inline app/src/viewmodels/home/MaximizeScreenViewModel.tsx:139]: would this make the final date a new date for each user that maxes screen?

**Review by EduardoNeville — CHANGES_REQUESTED** [review]
Looks good remove a few console.logs here and there and we can merge

File: app/src/viewmodels/home/MaximizeScreenViewModel.tsx (hunk 4 above)
Comment (Tristan-C789) [reply app/src/viewmodels/home/MaximizeScreenViewModel.tsx:139]: I just forced the type to be correct, it shouldn't change anything functionally speaking.

**Review by margdsq — APPROVED** [review]
Great PR ! Thank you for this

**Review by EduardoNeville — APPROVED** [review]

**Pull request merged**